### PR TITLE
Enable health metrics by default

### PR DESCRIPTION
**Pull request by randomanderson**

### CODE REVIEW

**Review by tylerbenson — APPROVED** [review]

**Review by richardstartin — APPROVED** [review]

**Comment by tylerbenson** [comment]
I suggest updating the pr name/description with more useful verbiage for the release notes.

**Comment by randomanderson** [comment]
This change breaks some CustomLogManager tests.  Investigating further

**Review by lpriima — APPROVED** [review]

**Comment by richardstartin** [comment]
@randomanderson I suspect this is because of DogStatsD's usage of `ScheduledExecutorService` creating conflicts with instrumentation. We should see if #1861 helps.

**Comment by richardstartin** [comment]
I suspect we may need to plug the executors used by JMXFetch/DogStatsD, since some of these are still failing with the same cause.

**Comment by tylerbenson** [comment]
This will likely be in a better state after #1958 is merged.

File: dd-trace-core/src/test/groovy/datadog/trace/core/CoreTracerTest.groovy [hunk 1]
```diff
@@ -43,36 +43,37 @@ class CoreTracerTest extends DDSpecification {
 
   def "verify defaults on tracer"() {
     when:
-    def tracer = CoreTracer.builder().build()
+    // FIXME this line should be unnecessary but a system property is leaking from somewhere
+    // Remove when config overrides is fixed
+    System.setProperty(PREFIX + HEALTH_METRICS_ENABLED, "true")
```
Comment (tylerbenson) [inline dd-trace-core/src/test/groovy/datadog/trace/core/CoreTracerTest.groovy:48]: perhaps this should clear the property instead of setting to true?

**Pull request merged**